main.py

```diff
@@ -604,7 +604,7 @@ def ui_access_wrapper(self, *args, **kwargs):
         topic_editor.TopicPublishSendMailHandler),
 
     get_redirect_route(
-        r'%s/<skill_id>' % feconf.CONCEPT_CARD_DATA_URL_PREFIX,
+        r'%s/<comma_separated_skill_ids>' % feconf.CONCEPT_CARD_DATA_URL_PREFIX,
```

Comment:
isn't comma(`,`) a reserver URI character? Also, I haven't seen any such pattern for URL path in real-world. can we move this to payloads? (not sure)

Reply:
> can we move this to payloads? (not sure)

Not sure about this as I don't know how we are going to write the acl.

@seanlip, do we have any URL of such a pattern in the codebase or do we follow this pattern? (I'm fine with keeping it  "comma-separated" as I don't see any issue with it except the url will be encoded (not readable, though it's not user-facing url) and I haven't seen a "," in a path before.)

Reply:
I agree with this @sophiewu6. Do commas get escaped? (Did you try this on a browser with more than 1 skill ID here?). Maybe we need to try a different separator.

Reply:
comma_sepataed_skill_ids is also used somewhere else in the codebase, and I have tested it with more than one skill and it worked. I believe it should be fine.

Reply:
Here's the general principle:

- GET handlers are for retrieving stuff, they should be read-only (i.e. calling them should not result in any datastore changes or other side-effects). With GET you will need to encode parameters in the URL, since there is no payload. I think $http.get() handles that, but just make sure there are backend + e2e tests covering at least one of these handlers so that we don't run into conversion problems at the interface. In this case I would leave comma_separated_skill_ids in the URL because there is no choice. (But it would not be part of the base URL, it should be an added parameter.)
- POST/PUT handlers are used for creating/modifying stuff respectively. They can change state in the datastore. Any requests to them should have data included in the json payload (including stuff like skill IDs, which can now be sent as a list of skill IDs). Try to refrain from using the URL for that unless it's something fairly generic (like exploration ID) and you're trying to keep it in sync with the corresponding GET handler so that they can all go in the same controller class (this makes sense in cases where the objects you're GETting are the same as the objects you're PUTting).

Does this help?

Reply:
I think this was tested manually by @sophiewu6 and it works fine. I agree that we should have backend and e2e tests for this scenario (in a separate PR). I just want to confirm that this resolves this comment so that we can move the PR forward in the review process?